geode-core/src/main/java/org/apache/geode/cache/query/internal/index/AbstractIndex.java

```diff
@@ -2102,4 +2105,14 @@ public void setPopulated(boolean isPopulated) {
   boolean isIndexOnPdxKeys() {
     return isIndexedPdxKeys;
   }
+
+  @Override
+  public synchronized boolean isMaintenanceOngoing() {
```

Comment:
Do we still need the method to be `synchronized` here, considering that the variable has already been declared as `volatile`?.

Reply:
Thanks for comment. Updated.